src/Microsoft.Framework.Logging.TraceSource/TraceSourceLoggerProvider.cs

```diff
@@ -97,5 +99,26 @@ private static bool HasDefaultSwitch(DiagnosticsTraceSource traceSource)
             return string.IsNullOrEmpty(traceSource.Switch.DisplayName) == string.IsNullOrEmpty(traceSource.Name) &&
                 traceSource.Switch.Level == SourceLevels.Off;
         }
+
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
+            {
+                if (disposing)
+                {
+                    _rootTraceListener.Flush();
+                    _rootTraceListener.Dispose();
+                }
+
+                _disposed = true;
+            }
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
Agree. 
 You recommend the same on following types ?
https://github.com/aspnet/Logging/pull/162/files#diff-f30224edfa4c3cede1d289dd6c0998eaR79
https://github.com/aspnet/Logging/pull/162/files#diff-4e0e02d9c6221bf5c7017a10fb867af3R43

And also removing comments on : 
https://github.com/aspnet/Logging/blob/dev/src/Microsoft.Framework.Logging/Logger.cs#L141